man/wleave.1.scd
wleave(1)

# NAME

wleave - A Wayland logout menu

# SYNOPSIS

*wleave* [options] [command]

# OPTIONS

*-h, --help*
	Show help message and stop

*-l, --layout* <layout>
	Specify a custom layout file, specify - to read the config from stdin

*--button-layout* "grid"
	Specify the way buttons should be laid out, "grid" being the default. Currently, only "grid" is supported.
	
*-D, --default-button* <button_label>
	Specify the label of the button to focus by default

*-S, --service*
	Run the application as a service, with all instances of wleave opening this one. Allows faster startup at the cost of running in the background.

*-v, --version*
	Show version number and stop

*-C, -css* <css>
	Specify a custom css file

*-b, --buttons-per-row* <num>
	Set the number of buttons per row, or use a fraction to specify the number of rows to be used (e.g. "1/1" for all buttons in a single row, "1/5" to distribute the buttons over 5 rows)

*-c, --column-spacing* <space>
	Set space between buttons columns

*-r, --row-spacing* <space>
	Set space between buttons rows

*-m, --margin* <padding>
	Set margin on all sides

*-L, --margin-left* <padding>
	Set margin for left of buttons

*-R, --margin-right* <padding>
	Set margin for right of buttons

*-T, --margin-top* <padding>
	Set margin for top of buttons

*-B, --margin-bottom* <padding>
	Set margin for bottom of buttons

*-A, --button-aspect-ratio* <ratio>
	Set the aspect ratio of the buttons, either as a float (as a number or string) or a ratio (e.g. "5/4").
	If unspecified, the buttons fill all available space between the margins.

*-f, --close-on-lost-focus*
	Closes the menu if focus is lost

*-k, --show-keybinds*
	Show the associated key binds for each button

*-p, --protocol* <protocol>
	Takes layer-shell, xdg, or none.
	The "layer-shell" allows transparency effects; however, only a few compositors correctly support it.
	The "xdg" protocol will work on almost all compositors, but does not allow for transparency.
	The "none" option shows the window as is, without full-screening it

*-d, --delay-command-ms* <number>
	The number of milliseconds to wait after an action before the associated command is executed

*-x, --no-version-info*
	Hide the version label

# DESCRIPTION

wleave is a Wayland-native logout script. It is a modern rewrite of Wlogout and a drop-in replacement.

# CONFIGURATION

wleave searches for a layout and style.css file in the following locations, in this order:

. $XDG_CONFIG_HOME/wleave/
. $XDG_CONFIG_HOME/wlogout/
. /etc/wleave/
. /etc/wlogout/
. /usr/local/etc/wleave
. /usr/local/etc/wlogout

If unset, $XDG_CONFIG_HOME defaults to *~/.config/*.

An error is raised when no layout file is found; However, the style.css file is optional. If you would like to customise either it is recommended that you copy the defaults from */etc/wleave/* into  *~/.config* and make any changes there.

# AUTHORS

Based on Wlogout by Haden Collins <collinshaden@gmail.com>. For more information about wlogout, see <https://github.com/ArtsyMacaw/wlogout>.
Rewrite by Natty <natty.sh.git@gmail.com>, see <https://github.com/AMNatty/wleave>.

# SEE ALSO

*wleave*(5)


completions/wleave.bash
_wleave() {
    local i cur prev opts cmd
    COMPREPLY=()
    if [[ "${BASH_VERSINFO[0]}" -ge 4 ]]; then
        cur="$2"
    else
        cur="${COMP_WORDS[COMP_CWORD]}"
    fi
    prev="$3"
    cmd=""
    opts=""

    for i in "${COMP_WORDS[@]:0:COMP_CWORD}"
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="wleave"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        wleave)
            opts="-v -s -l -D -C -b -c -r -m -L -R -T -B -A -d -f -k -p -x -h --version --service --layout --button-layout --default-button --css --buttons-per-row --column-spacing --row-spacing --margin --margin-left --margin-right --margin-top --margin-bottom --button-aspect-ratio --delay-command-ms --close-on-lost-focus --show-keybinds --protocol --no-version-info --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --service)
                    COMPREPLY=($(compgen -W "true false" -- "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -W "true false" -- "${cur}"))
                    return 0
                    ;;
                --layout)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -l)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --button-layout)
                    COMPREPLY=($(compgen -W "grid" -- "${cur}"))
                    return 0
                    ;;
                --default-button)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -D)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --css)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -C)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --buttons-per-row)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -b)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --column-spacing)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -c)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --row-spacing)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -r)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --margin)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -m)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --margin-left)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -L)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --margin-right)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -R)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --margin-top)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -T)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --margin-bottom)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -B)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --button-aspect-ratio)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -A)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --delay-command-ms)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -d)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --close-on-lost-focus)
                    COMPREPLY=($(compgen -W "true false" -- "${cur}"))
                    return 0
                    ;;
                -f)
                    COMPREPLY=($(compgen -W "true false" -- "${cur}"))
                    return 0
                    ;;
                --show-keybinds)
                    COMPREPLY=($(compgen -W "true false" -- "${cur}"))
                    return 0
                    ;;
                -k)
                    COMPREPLY=($(compgen -W "true false" -- "${cur}"))
                    return 0
                    ;;
                --protocol)
                    COMPREPLY=($(compgen -W "layer-shell none xdg" -- "${cur}"))
                    return 0
                    ;;
                -p)
                    COMPREPLY=($(compgen -W "layer-shell none xdg" -- "${cur}"))
                    return 0
                    ;;
                --no-version-info)
                    COMPREPLY=($(compgen -W "true false" -- "${cur}"))
                    return 0
                    ;;
                -x)
                    COMPREPLY=($(compgen -W "true false" -- "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

if [[ "${BASH_VERSINFO[0]}" -eq 4 && "${BASH_VERSINFO[1]}" -ge 4 || "${BASH_VERSINFO[0]}" -gt 4 ]]; then
    complete -F _wleave -o nosort -o bashdefault -o default wleave
else
    complete -F _wleave -o bashdefault -o default wleave
fi
